frontend/logic/preferences.js

```diff
@@ -73,3 +73,43 @@ export async function resubscribe(token) {
 		toast.error('Something went wrong. Please try again.');
 	}
 }
+
+export async function deleteAccount(token) {
+	try {
+		const client = new APIClient(token);
+		const response = await client.post('/api/delete-account');
+
+		if (response.ok) {
+			return true;
+		}
+
+		return false;
+	} catch (error) {
+		console.error(error);
+		toast.error('Something went wrong. Please try again.');
+	}
+}
+
+export async function cancelAccountDeletion(token) {
+	try {
+		const client = new APIClient(token);
+		const response = await client.post('/api/cancel-deletion');
+
+		if (response.ok) {
+			const { message } = await response.json();
+			await Router.push('/login');
+			toast.success(message, {
+				className: 'avenir bg-blue center pa3 lh-copy',
+			});
+			return;
+		}
+
+		const { errors } = await response.json();
+		toast.error(errors.message, {
```

Comment:
Rename `errors` to `error` or use an array in your backend.

Reply:
Unrelated to this PR. Will address later.

Reply:
Fixed in [e655bd5](https://github.com/meeshkan/vanity/commit/e655bd57d43a60778277e033d341a37dfbfb3bf6)